_rules/menuitem-non-empty-name-m6b1q3.md

```diff
@@ -0,0 +1,153 @@
+
+## Background
+
+- [Understanding Success Criterion 4.1.2: Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value)
+- [ARIA14: Using aria-label to provide an invisible label where a visible label cannot be used](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA14)
+- [ARIA16: Using aria-labelledby to provide a name for user interface controls](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA16)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+This element with the `menuitem` role has an [accessible name][] because of its text content.
+
+```html
+<ul role="menu">
+	<li role="menuitem">New file</li>
+</ul>
+```
+
+#### Passed Example 2
+
+This element with the `menuitem` role has an [accessible name][] because of its `aria-label` attribute.
+
+```html
+<link href="https://stackpath.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet" />
+<ul role="menu">
+	<li role="menuitem" aria-label="New file">
+		<i class="fa fa-file"></i>
+	</li>
+</ul>
+```
+
+#### Passed Example 3
+
+This element with the `menuitem` role has an [accessible name][] because its `aria-labelledby` attribute references a `div` element with text content.
+
+```html
+<link href="https://stackpath.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet" />
+<ul role="menu">
+	<li role="menuitem" aria-labelledby="newfile">
+		<i class="fa fa-file"></i>
+		<span id="newfile" hidden>New file</span>
+	</li>
+</ul>
+```
+
+#### Passed Example 4
+
+This element with the `menuitem` role has an [accessible name][] because of its `title` attribute.
+
+```html
+<link href="https://stackpath.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet" />
+<ul role="menu">
+	<li role="menuitem" title="New file">
+		<i class="fa fa-file"></i>
+	</li>
+</ul>
+```
+
+### Failed
+
+#### Failed Example 1
+
+This element with the `menuitem` role has no [accessible name][] because it has no content or attribute that can provide it.
+
+```html
+<link href="https://stackpath.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet" />
+<ul role="menu">
+	<li role="menuitem">
+		<i class="fa fa-file"></i>
+	</li>
+</ul>
+```
+
+#### Failed Example 2
+
+This element with the `menuitem` role has no [accessible name][] because it has no content or attribute that can provide it.
+
+```html
+<html lang="en">
+	<style>
+		.offscreen {
+			position: absolute;
+			left: -9999px;
+			top: -9999px;
+		}
+	</style>
+	<link href="https://stackpath.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet" />
+	<ul role="menu" class="offscreen">
+		<li role="menuitem">
+			<i class="fa fa-file"></i>
+		</li>
+	</ul>
+</html>
+```
+
+### Inapplicable
```

Comment:
For coverage of the Applicability, we need an example with a non-`menuitem` role.